### ISSUE TITLE
caller is wrong about packages declared in a sub

### ISSUE THREAD

**Issue by toddr** [CpanelInc/perl-compiler#33]
```
$>perlcc -r -e 'sub c { caller(0) }  sub foo { package PQR; main->c() } print((foo())[0]); print "\n";'
main
$>perl -e 'sub c { caller(0) }  sub foo { package PQR; main->c() } print((foo())[0]); print "\n";'  
PQR
```

**Comment by toddr** [comment]
Documented failure in 929db5c9895f020230f2eadb7a9e5c9d2f4fae51

**Comment by toddr** [comment]
We're probably still missing something CV.pm related.

**Comment by toddr** [comment]
We were clearing empty packages from the white list. You can't do that.

Fixed by daafd1190615d8fc72f19a280400c23ee4a87cba
